### PR TITLE
Slightly Simplified STP Logic

### LINKED ISSUE #738: parameter object initialization order issue
### Description of the task
Provide a way to initialize parameters, instead of relying on static initialization.

### Outline the task in more detail
Since we just use namespaces and don't have a singleton or class structure, and because c++ doesn't guarantee any order of initialization, there is no guarantee parameters will be ready when other constructors/objects need them at the very beginning of the program.

So in the functions `value` and `setValue`, we should only be returning the value of the parameter in the registry, not the value of the parameter instance used to access the registry. (We currently do so, so if any objects are initialized before the parameters are initialized, we return T() as there is nothing in the registry during this time.)

### Acceptance criteria

- [ ] Figure out a way and fix initialization order to make sure parameters get initialized first
- [ ] Remove return this->value and throw exception instead
- [ ] Unit tests

### LINKED ISSUE #783: Refactor stp.cpp
### Description of the task

`stp.cpp / stp.h` are the files responsible for selecting plays and assigning tactics during our autonomous gameplay, and as such are very critical to be easy to understand and maintain.

Right now this file has several very large / long functions, and some fairly deep nested blocks of loops and conditions. This should all be split out into smaller, more manageable functions so components can be tested separately and the code is easier to read. The very large functions are currently hard to follow.

### Acceptance criteria

- [ ] Split the large functions in stp.cpp into smaller functions that perform single tasks. This should also reduce the nested code
- [ ] Write new tests for the smaller functions
- [ ] Update existing tests

### PR DESCRIPTION
**Pull request by garethellis0**
### Testing Done
Tests pass!

### Resolved Issues
resolves #783, there wasn't as many wins here as we expected when we originally wrote this ticket, but I think this is because of a few small simplifications that got made via other PR's since then.

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

**Comment by MathewMacDougall** [comment]
I think you referenced the wrong ticket?

**Comment by garethellis0** [comment]
> I think you referenced the wrong ticket?

Whoops, fixed.

**Review by MathewMacDougall — APPROVED** [review]

**Review by jordanzliu — APPROVED** [review]
💯

**Review by jonathanlew — APPROVED** [review]
:+1:
